### PR TITLE
docs(fbc): add olm.constraint and compound constraint schemas

### PR DESCRIPTION
**Pull request by estroz**
This PR adds schemas for `olm.constraint` and compound constraints from their respective EP's, linked in the doc.

Signed-off-by: Eric Stroczynski <ericstroczynski@gmail.com>

### CODE REVIEW

File: content/en/docs/Reference/file-based-catalogs.md [hunk 1]
```diff
@@ -333,6 +341,166 @@ The `olm.gvk.required` property [cue][cuelang-spec] schema is:
...
+      packageName: "foo"
+      versionRange: ">=1.0.0"
+    }
+  }
+}
+```
+
+##### `#Cue`
+
+TODO
```
Comment (estroz) [inline content/en/docs/Reference/file-based-catalogs.md:425]: @dinhxuanvu this needs a follow-up with a cue schema and short description with link to the EP.

**Comment by estroz** [comment]
@dinhxuanvu @perdasilva I rebased and wrote on top of #220 here. Either I can rebase after #220 is merged or I'll give @dinhxuanvu access to my fork so he can.

/hold

**Comment by dinhxuanvu** [comment]
Consolidate this PR to #220

**Pull request closed without merging**